### PR TITLE
[BOUNTY] Adds advanced injectors to genetics

### PR DESCRIPTION
**Pull request by Time-Green**
@PKPenguin321 
Adds Advanced DNA Injectors to genetics
You can now make selections of mutations that you want to put in one injector. The recharge is 1.6x longer than normal injectors.
The amount of mutations is limited to 50 instability or 10 mutations.
image
(The awkward spacing was fixed and the injector names were bolded)
:cl:
add: Adds advanced DNA injectors to genetics. You can now put multiple mutations in an injector
/:cl:

### CODE REVIEW

**Comment by drderp3635** [comment]
Doesn't this mean you can stack every disability onto one syringe if it is limited by instability? 
Would that be a power concern?

**Comment by knacker48** [comment]
Yeah you may want to put a max amount of mutations per injector. Maybe 5 or 6?

**Comment by nemvar** [comment]
Maybe some interaction with machine upgrades?

File: code/game/machinery/computer/dna_console.dm [hunk 1]
```diff
@@ -826,6 +883,46 @@
 						to_chat(usr, "<span class='notice'>You apply [CM] to [HM.name].</span>")
 						stored_chromosomes -= CM
 						CM.apply(HM)
+		if("expand_advinjector")
```
Comment (nemvar) [inline code/game/machinery/computer/dna_console.dm]: Does this thing check if the selected mutation is already in the injector?

Comment (Time-Green) [reply code/game/machinery/computer/dna_console.dm]: It doesn't. But there's just no ill effects since it just doesn't do anything.

**Comment by PKPenguin321** [comment]
I figure that as the bounty author I should put some reasoning for this PR.

Genetics is currently basically useless for sabotage. Old genetics had the ability to save entire genomes as SE Injectors, allowing you to give literally every mutation in one injector. This PR isn't quite that, but that made genetics useful for weaponizing or using in sneaky, malicious ways. Current genetics can only transfer one mutation per injector, meaning the most sabotage you'll see from it will boil down to "I printed a monkey injector and injected you with it."

Current genetics has a lot more disabilities than old genetics, so I didn't want these to allow you to give all mutations at once; hence the 10 mutation cap. At the same time, old genetics was way too efficient at giving out powers, with a single injector giving you literally all of them. With the instability limit (and extra printer cooldown) in this PR, you can still give some powers, but definitely not all: If you put Hulk into an injector, for example, you can only fit 10 instability worth of powers, and none of the actually good powers will fit.

So yeah, tl;dr a geneticist who puts in effort to discover a lot of disabilities and build their own SE syringe is threatening again, as with old genetics. Current genetics sux because there's no way to use it in a malicious way with the best you can do being an injector with Monkey in it. Also you can rename these syringes so you can place "Epic powers combo" on the floor and some chump can use it and give themselves Tourette syndrome and blindness.

File: code/game/machinery/computer/dna_console.dm [hunk 2]
```diff
@@ -36,11 +36,21 @@
...
+	///Amount of mutations we can store
 	var/list/buffer[NUMBER_OF_BUFFERS]
+	///mutations we have stored
 	var/list/stored_mutations = list()
+	///chromosomes we have stored
 	var/list/stored_chromosomes = list()
+	///combinations of injectors for the 'injector selection'. format is list("Elsa" = list(Cryokinesis, Geladikinesis), "The Hulk" = list(Hulk, Gigantism), etc) Glowy and the gang being an initialized datum
+	var/list/injector_selection = list()
+	///max amount of selections you can make
+	var/max_injector_instability = 2
```
Comment (PKPenguin321) [inline code/game/machinery/computer/dna_console.dm]: I feel like the names on these are reversed?

File: code/game/machinery/computer/dna_console.dm [hunk 3]
```diff
@@ -36,11 +36,21 @@
 	var/combine
 	var/radduration = 2
 	var/radstrength = 1
+	///the max instability of the advanced injector.
+	var/advanced_injector_limit = 50
```
Comment (PKPenguin321) [inline code/game/machinery/computer/dna_console.dm]: I feel like the names on these are reversed?

File: code/game/machinery/computer/dna_console.dm (hunk 2 above)
Comment (Time-Green) [reply code/game/machinery/computer/dna_console.dm]: I have absolutely no idea how that happened, I distinctly remember not doing this

**Comment by actioninja** [comment]
Not 100% sold on this but if it ends up being a problem the numbers on limits can be tweaked or it can be reworked.

**Pull request merged**